materials/variation_with_state_test.py

```diff
@@ -0,0 +1,278 @@
+        result = state_model.query_value({'exposure time': 0, 'temperature': 2.5}, method='linear')
+        self.assertAlmostEqual(result, 6.5)
+        result = state_model.query_value({'exposure time': 0.05, 'temperature': 2}, method='linear')
+        self.assertAlmostEqual(result, (2**2 + 2.5**2)/2)
+
+        # Check that a scalar query returns a scalar
+        result = state_model.query_value({'exposure time': 0, 'temperature': 1})
+        self.assertFalse(hasattr(result, '__len__'))
+
+        # Check query of multiple points
+        result = state_model.query_value({'exposure time': [0, 0.05], 'temperature': [1, 2]})
+        self.assertTrue(hasattr(result, '__len__'))
+        self.assertEqual(result[0], 1**2)
+        self.assertAlmostEqual(result[1], (2**2 + 2.5**2)/2)
+
+        # Check bad query
+        with self.assertRaises(ValueError):
+            # fish is not a state variable.
+            state_model.query_value({'fish': 1., 'temperature': 1})
+        with self.assertRaises(ValueError):
+            # fish is not a state variable.
+            state_model.query_value({'fish': 1., 'exposure time': 1})
+
+    def test_domain_1d(self):
+        """Test get_state_domain on a 1-d lookup table."""
+        # Setup
+        state_model = vstate.VariationWithStateTable(
+            ['temperature'], {'temperature': 'kelvin'}, 'multiplier', 'reference',
+            np.arange(4), np.arange(4)**2,
+            ['linear'])
+
+        # Action
+        domain = state_model.get_state_domain()
+
+        # Verification
+        self.assertEqual(len(domain), 1)
+        self.assertTrue('temperature' in domain)
+        self.assertEqual(domain['temperature'][0], 0)
+        self.assertEqual(domain['temperature'][1], 3)
+
+
+    def test_domain_2d(self):
+        """Test get_state_domain on a 2-d lookup table."""
+        # Setup
+        yaml_dict = {
+            'exposure time': {
+                0.0: {
+                    'temperature': np.arange(4),
+                    'values': np.arange(4)**2
+                },
+                0.1: {
+                    'temperature': np.arange(4),
+                    'values': (np.arange(4) + 0.5)**2
+                },
+            }
+        }
+        state_vars = ['exposure time', 'temperature']
+        scales = ['linear', 'linear']
+        #pylint: disable=protected-access
+        interp_points, interp_values = vstate._create_interp_arrays_from_yaml_table(
+            yaml_dict, state_vars, scales)
+
+        state_model = vstate.VariationWithStateTable(
+            state_vars, {'exposure time': 'hour', 'temperature': 'kelvin'}, 'multiplier', 'reference',
+            interp_points, interp_values,
+            scales)
+
+        # Action
+        domain = state_model.get_state_domain()
+
+        # Verification
+        self.assertEqual(len(domain), 2)
+        self.assertTrue('exposure time' in domain)
+        self.assertTrue('temperature' in domain)
```

Comment:
`assertIn` is available for meganits

Reply:
Ooohh nifty. Switched to `assertIn` in 28a3c2e226383d2fb7bcf83f20fdc00876bc27c0